src/socket.rs

```diff
@@ -369,6 +369,49 @@ impl Socket {
         self.inner.set_ttl(ttl)
     }
 
+    /// Gets the value for the `SO_MARK` option on this socket.
+    ///
+    /// This value gets the socket mark field for each packet sent through
+    /// this socket.
+    ///
+    /// This function is only available on Unix and requires the
+    /// `CAP_NET_ADMIN` capability.
+    #[cfg(target_os = "linux")]
+    pub fn mark(&self) -> io::Result<u32> {
```

Comment:
Since this Unix (well Linux) only API please move it to `src/sys/unix.rs`, e.g. see https://github.com/rust-lang/socket2-rs/blob/dc82e662cda027f69abb75917777943fbffb8463/src/sys/unix.rs#L377-L417